src/tree-selector/treeSelectorComponent.ts

```diff
@@ -0,0 +1,69 @@
+import * as ng from 'angular';
+
+export class TreeSelectorController {
+
+  public initialData: any;
+  public selectables: any;
+  public openTreeView: any;
+  private reg: any;
+  private url: string;
+  private handleSelected: any;
+
+  /*@ngInject*/
+  constructor(private $http, private $timeout : ng.ITimeoutService){}
+
+  public $onInit() {
+    this.initialData = this.parseSelectable(this.initialData);
+    console.log('selectables: ', this.selectables, this.initialData);
+  }
+
+  private getData = (id?) => {
+    return this.$http({
+      method: 'GET',
+      url: id ? `${this.url}?id=${encodeURIComponent(id)}` : this.url,
+    }).then((response) => {
+      return this.parseSelectable(response.data);
+    });
+  }
+
+  public lazyLoad = (node) => {
+    const dataPromise = this.getData(node.key);
+    return dataPromise.then((data) => {
+      return this.parseSelectable(data);
+    });
+  }
+
+  public parseSelectable = (data: any) => {
+    return data.map((node, key) => {
+      const parsedData = {...node};
+      if(parsedData.nodes) {
+        parsedData.nodes = node.nodes.map((childNode, childKey) => {
```

Comment:
Is it guaranteed that we're receiving just one level of children? What if the tree builder or its current developer (me) goes crazy and sends you all the node hierarchy in some special cases. I'm thinking about a tree like:
```
* grandpa
|
** dad
 |
 ** me
  |
  ** my son
```
The `TreeBuilder` *might know* that this would be a waste of resources if there is just a single children on every level so it *might send* the whole subtree when expanding `grandpa` instead of having multiple requests. 

This is just and idea, but feel free to stop me :wink: